cloudpathlib/azure/azblobclient.py

```diff
@@ -202,7 +202,7 @@ def _check_hns(self, cloud_path: AzureBlobPath) -> Optional[bool]:
             try:
                 account_info = self.service_client.get_account_information()  # type: ignore
                 self._hns_enabled = account_info.get("is_hns_enabled", False)  # type: ignore
-            except ResourceNotFoundError:
+            except (ResourceNotFoundError, HttpResponseError):
```

Comment:
Per the stack trace in #486, can we confirm the error code is `ErrorCode:AuthorizationPermissionMismatch` and if not, re-raise the `HttpResponseError`? That will make other potential errors easier to track down.

Reply:
Is checking for `HTTPStatus.FORBIDDEN` sufficient?